server/src/main/java/org/elasticsearch/index/seqno/RetentionLeaseStats.java

```diff
@@ -0,0 +1,111 @@
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * Represents retention lease stats.
+ */
+public final class RetentionLeaseStats implements ToXContentFragment, Writeable {
+
+    private final Collection<RetentionLease> leases;
+
+    /**
+     * The underlying retention leases backing this stats object.
+     *
+     * @return the leases
+     */
+    public Collection<RetentionLease> leases() {
+        return leases;
+    }
+
+    /**
+     * Constructs a new retention lease stats object from the specified leases.
+     *
+     * @param leases the leases
+     */
+    public RetentionLeaseStats(final Collection<RetentionLease> leases) {
+        this.leases = Objects.requireNonNull(leases);
+    }
+
+    /**
+     * Constructs a new retention lease stats object from a stream. The retention lease stats should have been written via
+     * {@link #writeTo(StreamOutput)}.
+     *
+     * @param in the stream to construct the retention lease stats from
+     * @throws IOException if an I/O exception occurs reading from the stream
+     */
+    public RetentionLeaseStats(final StreamInput in) throws IOException {
+        leases = in.readList(RetentionLease::new);
+    }
+
+    /**
+     * Writes a retention lease stats object to a stream in a manner suitable for later reconstruction via
+     * {@link #RetentionLeaseStats(StreamInput)} (StreamInput)}.
+     *
+     * @param out the stream to write the retention lease stats to
+     * @throws IOException if an I/O exception occurs writing to the stream
+     */
+    @Override
+    public void writeTo(final StreamOutput out) throws IOException {
+        out.writeCollection(leases);
+    }
+
+    /**
+     * Converts the retention lease stats to {@link org.elasticsearch.common.xcontent.XContent} using the specified builder and pararms.
+     *
+     * @param builder the builder
+     * @param params  the params
+     * @return the builder that these retention leases were converted to {@link org.elasticsearch.common.xcontent.XContent} into
+     * @throws IOException if an I/O exception occurs writing to the builder
+     */
+    @Override
+    public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
+        builder.startObject("retention_leases");
+        {
+            builder.startArray("leases");
+            {
+                for (final RetentionLease retentionLease : leases) {
+                    builder.startObject();
+                    {
+                        builder.field("id", retentionLease.id());
+                        builder.field("retaining_sequence_number", retentionLease.retainingSequenceNumber());
```

Comment:
perhaps `retaining_seq_no` given that we already use this abbreviation in other seq-no related stats output

Reply:
I pushed 2429d7e9da4137470db191e156756cb59a26d4c9.